.changeset/silly-cars-double.md

```diff
@@ -0,0 +1,9 @@
+---
+"@pnpm/resolve-dependencies": patch
+"@pnpm/resolver-base": patch
+"@pnpm/npm-resolver": patch
+"@pnpm/core": patch
+"pnpm": patch
+---
+
+When resolving dependencies, prefer versions that are already used in the root of the project. This is important to minimize the number of packages that will be nested during hoisting [#6054](https://github.com/pnpm/pnpm/pull/6054).
```

Comment:
This change prefers selectors used by the any importer (or internal package), right? The phrasing "_in the root of the project_" might imply this only affects the root `package.json`?

Reply:
> This change prefers selectors used by the any importer 

yes